modules/openapi-generator/src/main/resources/typescript/model/model.mustache

```diff
@@ -62,7 +62,9 @@ export class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{
         {{/discriminatorValue}}
         {{/allVars}}
         {{#discriminatorName}}
+        {{^discriminator.isEnum}}
```

Comment:
this is the meaningful change of this PR

Reply:
The discriminator is the `type` and `type` is of type `InteractionTypeEnum`, thus the template string would evaluate to `true`, hence with the template negation, it would not emit the guarded class name assignment.

Reply:
thanks for the explanation! I wonder why there even is such a default value assigned, do you happen to know?

Reply:
I do not, sorry, I am assuming it is a remnant from making sure each class has a sane default value to discriminate on, however the implementation of it doesn't make sense. It works fine when the discriminator is of string type, which is the majority of implementations I've seen, but it obviously doesn't work any more as soon as it's a fixed enum.